### PR TITLE
Update common-jvm to 0.31.0.

### PR DESCRIPTION
**Pull request by SanjayVas**
This also addresses some Maven dependency issues.

---
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/278)

### CODE REVIEW

**Review by stevenwarejones — COMMENTED** [review]
did you test the maven publish?

> Reviewed 11 of 12 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/278#-:-Mx2LgtF00fkKqwtXTij:b-c5wezp)* status: 11 of 12 files reviewed, all discussions resolved (waiting on @stevenwarejones)

File: src/main/kotlin/org/wfanet/panelmatch/client/deploy/example/gcloud/BUILD.bazel [hunk 1]
```diff
@@ -23,18 +23,12 @@ kt_jvm_library(
 java_binary(
     name = "GoogleCloudExampleDaemonMain",
     main_class = "org.wfanet.panelmatch.client.deploy.example.gcloud.GoogleCloudExampleDaemonMainKt",
-    runtime_deps = [
-        ":gcloud",
-        "//imports/java/com/google/common:tink_gcpkms",
-    ],
+    runtime_deps = [":gcloud"],
```
Comment (yunyeng) [inline src/main/kotlin/org/wfanet/panelmatch/client/deploy/example/gcloud/BUILD.bazel:26]: Do we confirm tink is running without runtime_deps? That was not the case when I was deploying.

**Review by yunyeng — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/278)* status: 11 of 12 files reviewed, all discussions resolved (waiting on @stevenwarejones)

**Review by SanjayVas — COMMENTED** [review]
The `//src/main/kotlin/org/wfanet/panelmatch/client/deploy:maven.publish` target builds successfully.

This mainly drops explicit version specifications, but doesn't change what's actually being pulled (aside from Tink, which shouldn't be pulled from Maven at all as we build it from source).

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/278#-:-Mx6-LtaDwAmVTaIlrVb:bbklixf)* status: 11 of 12 files reviewed, all discussions resolved (waiting on @stevenwarejones)

**Review by stevenwarejones — APPROVED** [review]
> Reviewed 1 of 12 files at r1.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/278)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @SanjayVas)

**Review by SanjayVas — COMMENTED** [review]
> Reviewed 11 of 12 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/278)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @SanjayVas)

**Review by SanjayVas — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/278)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @SanjayVas)

**Pull request merged**